docs-ref-conceptual/python-sdk-azure-authenticate.md

```diff
@@ -23,36 +23,53 @@ Several options are available to authenticate your application with Azure when u
 
 Store the credentials securely in a configuration file, the registry, or Azure KeyVault. It is recommended to use [ADAL](https://github.com/AzureAD/azure-activedirectory-library-for-python)
 and the SDK ADAL wrapper. Please refer to the ADAL website for all the available scenarios
-list and samples. For instance for service principal authentication:
+list and samples.
+
+The following example uses a [Service Principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli?toc=%2fazure%2fazure-resource-manager%2ftoc.json) for authentication.
+
+> [!NOTE]
+> You can create a Service Principal via the Azure CLI 2.0
+> ```bash
+> az ad sp create-for-rbac --name "MY-PRINCIPAL-NAME" --password "STRONG-SECRET-PASSWORD"
+> ```
 
 ```python
     import adal
     from msrestazure.azure_active_directory import AdalAuthentication
+    from msrestazure.azure_cloud import AZURE_PUBLIC_CLOUD
+
+    # Tenant ID for your Azure Subscription
+    TENANT_ID = 'ABCDEFGH-1234-1234-1234-ABCDEFGHIJKL'
 
-    login_endpoint = 'https://login.microsoftonline.com/'
-    tenant = 'ABCDEFGH-1234-1234-1234-ABCDEFGHIJKL'
-    resource = 'https://management.core.windows.net/' #AAD graph resource
+    # Your Service Principal App ID
+    CLIENT = 'a2ab11af-01aa-4759-8345-7803287dbd39'
 
-    client = 'a2ab11af-01aa-4759-8345-7803287dbd39'
-    key = 'password'
+    # Your Service Principal Password
+    KEY = 'password'
 
-    context = adal.AuthenticationContext(LOGIN_ENDPOINT+TENANT)
+    LOGIN_ENDPOINT = AZURE_PUBLIC_CLOUD.endpoints.active_directory
+    RESOURCE = AZURE_PUBLIC_CLOUD.endpoints.active_directory_resource_id
+
+    context = adal.AuthenticationContext(LOGIN_ENDPOINT + '/' + TENANT_ID)
     credentials = AdalAuthentication(
         context.acquire_token_with_client_credentials,
-        resource,
-        client,
-        key
+        RESOURCE,
+        CLIENT,
+        KEY
     )
 ```
 
-All ADAL valid calls can be used with the `AdalAuthentication` class.
+> [Note!]
+> To connect to one of the Azure sovereign clouds, import the relevant cloud environment instead of `AZURE_PUBLIC_CLOUD`, e.g. `AZURE_CHINA_CLOUD` or `AZURE_GERMAN_CLOUD`.
+
+All valid ADAL calls can be made with the `AdalAuthentication` class.
 
-Then create a client object to start working with the API:
+Next, create a client object to start working with the API:
 
 ```python
 from azure.mgmt.compute import ComputeManagementClient
 
-# change for your actual subscription id
+# Your Azure Subscription ID
 subscription_id = '33333333-3333-3333-3333-333333333333'
 
 client = ComputeManagementClient(credentials, subscription_id)
```

Comment:
Add a note, that to connect to sovereign clone, base_url on the client should be updated to the RESOURCE value used earlier.